bin/lesson_check.py

```diff
@@ -257,37 +256,56 @@ def read_all_markdown(source_dir, parser):
     {path : {'metadata':yaml, 'metadata_len':N, 'text':text, 'lines':[(i, line, len)], 'doc':doc}}
     """
 
+    result = {}
+    for d in SOURCE_DIRS:
+        dpath = os.path.join(source_dir, d)
+
+        pattern = os.path.join(dpath, '*.md')
+        for filename in glob.glob(pattern):
+            data = read_markdown(parser, filename)
+            if data:
+                result[filename] = data
+
+    return result
+
```

Comment:
This is an unconditional `return`, so the rest of the function becomes unreachable after this line

Reply:
Sorry, this is actually a leftover from what I was working on. The code below is the old implementation. I'll remove it.